src/pages/workspaces/workspace/WorkspaceContainer.ts

```diff
@@ -258,37 +258,39 @@ const WorkspaceAccessError = () => {
   ]);
 };
 
-interface WrapWorkspaceProps {
+export interface WrapWorkspaceOptions {
   breadcrumbs: (props: { name: string; namespace: string }) => ReactNode[];
   activeTab?: string;
   title: string;
 }
 
-interface WrappedComponentProps extends ComponentPropsWithRef<any> {
+export interface WrappedComponentProps {
+  ref: Ref<{ refresh: () => void }>;
+  namespace: string;
+  name: string;
   workspace: Workspace;
   refreshWorkspace: () => void;
   analysesData: AppDetails & CloudEnvironmentDetails;
   storageDetails: StorageDetails;
 }
 
-type WrappedWorkspaceComponent<T extends WrappedComponentProps> = (props: T) => ReactNode;
+export type WrappedWorkspaceComponent = (props: WrappedComponentProps) => ReactNode;
 
-type WorkspaceWrapperFunction<T extends WrappedComponentProps> = (
-  component: WrappedWorkspaceComponent<T>
-) => WrappedWorkspaceComponent<T>;
+export interface WorkspaceWrapperProps {
+  namespace: string;
+  name: string;
+}
 
 /**
  * wrapWorkspaces contains a component in the WorkspaceContainer
  * and provides the workspace analysesData and storageDetails
  * */
-export const wrapWorkspace = <T extends WrappedComponentProps>(
-  props: WrapWorkspaceProps
-): WorkspaceWrapperFunction<T> => {
-  const { breadcrumbs, activeTab, title } = props;
-  return (WrappedComponent: WrappedWorkspaceComponent<T>): WrappedWorkspaceComponent<T> => {
-    const Wrapper = (props) => {
+export const wrapWorkspace = (opts: WrapWorkspaceOptions) => {
+  const { breadcrumbs, activeTab, title } = opts;
```

Comment:
@cahrens also renaming `props` to `opts` here avoided a shadowed variable. Previously, Wrapper's `props` shadowed the `props` passed to `wrapWorkspace`.